File: practica 1/sesion 2/DS_P1S2-FactoriaAbstracta-MetodoFactoriaJava/src/ds_p1_sesion2/Cliente.java
package ds_p1_sesion2;
import java.util.Vector;

public class Cliente {
    public Vector<Trabajo> trabajos = new Vector<Trabajo>();
    public String nombre;
    public float importancia;

    public Cliente(String nombre) {
        this.nombre = nombre;
    }

    @Override
    public String toString() {
        return "Cliente{" + "trabajos=" + trabajos + ", nombre=" + nombre + ", importancia=" + importancia + '}';
    }

    public Vector<Trabajo> getTrabajos() {
        return trabajos;
    }

    public void setTrabajos(Vector<Trabajo> trabajos) {
        this.trabajos = trabajos;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public float getImportancia() {
        return importancia;
    }

    public void setImportancia(float importancia) {
        this.importancia = importancia;
    }

    public void crearTrabajo(String nombre, float presupuesto) {
        Trabajo trabajoCreado = new Trabajo(this, nombre, presupuesto);
        trabajos.add(trabajoCreado);
    }

    public void editarTrabajo() {
        throw new UnsupportedOperationException();
    }

    public void cancelarTrabajo(Trabajo aCancelar) {
        int indice = trabajos.indexOf(aCancelar);
        trabajos.remove(indice);
    }
}

File: practica 1/sesion 2/DS_P1S2-FactoriaAbstracta-PrototipoRuby/Cliente.rb
require('./Trabajo.rb');

module DSPrototype
public
class Cliente

	protected
  def clone
      return Cliente.new(@nombre,@importancia)
  end

	private
	def initialize(nombre)
		# @AttributeType Trabajo
		# # @AssociationType Trabajo
		# # @AssociationMultiplicity 1..*
		@trabajos = Array.new()
		@nombre = nombre
		@importancia = 1
	end
	
	public
	def to_string()
	  return "Cliente > [Nombre : #{@nombre} ; Importancia : #{@importancia} ; Trabajos creados : #{@trabajos} ]"
	end
	
	def crear_trabajo(descripcion, presupuesto)
	  trabajos.push(Trabajo.new(self,descripcion,presupuesto))
	end
	
	
	attr_accessor :trabajos, :nombre, :importancia
end
end
